source/MRViewer/MRSurfaceManipulationWidget.cpp

```diff
@@ -15,20 +15,21 @@
 #include <MRMesh/MRMeshRelax.h>
 #include <MRMesh/MRBitSetParallelFor.h>
 
+
 namespace MR
 {
+//const float k = r < 1-a ? std::sqrt( sqr( 1 - a ) - sqr( r ) ) + ( 1 - a ) : -std::sqrt( sqr( a ) - sqr( r - 1 ) ) + a; // alternative version F_point_shift(r,i) (i == a)
 
 void SurfaceManipulationWidget::init( const std::shared_ptr<ObjectMesh>& objectMesh )
 {
     obj_ = objectMesh;
     diagonal_ = obj_->getBoundingBox().diagonal();
-    settings_.radius = diagonal_ / 100.f;
+    minRadius_ = obj_->avgEdgeLen() * 2.f;
```

Comment:
Should widget be aware of radius limits?

Reply:
I think it is. Radius less than edge doesn't make sense